src/redist/targets/GenerateMSIs.targets

```diff
@@ -283,49 +280,36 @@
     </ItemGroup>
   </Target>
 
-  <Target Name="GenerateManifestsMsi"
-        DependsOnTargets="GenerateLayout;AcquireWix;MsiTargetsSetupInputOutputs;SetSdkBrandingInfo"
-        Condition="$(ProductMonikerRid.StartsWith('win')) And '$(Architecture)' != 'arm'"
-        Inputs="@(BundledManifests);$(ManifestsGenerateMsiPowershellScript)"
-        Outputs="$(ManifestsMSIInstallerFile)">
-
+  <Target Name="GenerateWorkloadManifestsWxs">
     <PropertyGroup>
-      <ManifestsGenerateMsiPowershellScript>$(MSBuildThisFileDirectory)packaging/windows/clisdk/generatemanifestsmsi.ps1</ManifestsGenerateMsiPowershellScript>
+      <WorkloadManifestsWxsPath>$(IntermediateOutputPath)WorkloadManifests.wxs</WorkloadManifestsWxsPath>
+      <WorkloadManifestsWxsContent>
+<![CDATA[
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. -->
+<Wix xmlns="http://schemas.microsoft.com/wix/2006/wi">
+    <Fragment>
+        <PackageGroup Id="PG_WorkloadManifests">
+            @(BundledManifests->'<MsiPackage SourceFile="%(RestoredMsiPathInNupkg)">
+                <MsiProperty Name="DOTNETHOME" Value="[DOTNETHOME]" />
+                <MsiProperty Name="ALLOWMSIINSTALL" Value="True" />
```

Comment:
You should be able to remove the ALLOWMSIINSTALL property - we don't generate that for the manifest installer